### PR TITLE
[O2B-1119] Add QC flag fetch endpoints

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Added QC flag fetch by id and fetch all endpoint

Changes made to the database:
- NA

### CODE REVIEW

File: test/api/qcFlags.test.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+const { expect } = require('chai');
+const request = require('supertest');
+const { server } = require('../../lib/application');
+const { resetDatabaseContent } = require('../utilities/resetDatabaseContent.js');
+
+module.exports = () => {
+    before(resetDatabaseContent);
+    describe('GET /api/qcFlags/:id', () => {
+        it('should successfuly fetch one QC flag type', async () => {
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:22]: Not QC flag TYPE

File: test/api/qcFlags.test.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            const response = await request(server).get('/api/qcFlags/4');
+            expect(response.status).to.be.equal(200);
+            const { data: qcFlag } = response.body;
+            expect(qcFlag).to.be.eql({
+                id: 4,
+                from: (1647914400 + 10000) * 1000,
+                to: (1647914400 + 10000) * 1000,
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:29]: Please put dates

File: test/api/qcFlags.test.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                comment: 'Some qc comment 4',
+                createdAt: 1707825439000,
+                updatedAt: 1707825439000,
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:32]: Same

File: test/api/qcFlags.test.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                    status: 404,
+                    title: 'Not found',
+                    detail: 'Quality Control Flag with this id (99999) could not be found',
+                },
+            ]);
+        });
+    });
+
+    describe('GET /api/qcFlags', () => {
+        it('should successfuly fetch all data', async () => {
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:57]: Could you put more explicit message?

File: test/api/qcFlags.test.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(5);
+            const oneFetchedFlag = data.find(({ id }) => id === 4);
+            expect(oneFetchedFlag).to.be.eql({
+                id: 4,
+                from: (1647914400 + 10000) * 1000,
+                to: (1647914400 + 10000) * 1000,
+                comment: 'Some qc comment 4',
+                createdAt: 1707825439000,
+                updatedAt: 1707825439000,
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:71]: Dates

File: test/api/qcFlags.test.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+    describe('GET /api/qcFlags', () => {
+        it('should successfuly fetch all data', async () => {
+            const response = await request(server).get('/api/qcFlags');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 5, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(5);
+            const oneFetchedFlag = data.find(({ id }) => id === 4);
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:64]: Please check the ids of all, no need to check all the properties I think

File: test/api/qcFlags.test.js [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+                .get('/api/qcFlags?filter[simulationPassIds][]=1&filter[runNumbers][]=106&filter[dplDetectorIds][]=1');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 1, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(1);
+            expect(data[0].id).to.be.equal(5);
+        });
+
+        it('should retrive no records when filtering on ids', async () => {
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:111]: filtering on non-existing ids

I would say that you can merge this with the other filtering on id, up to you

File: test/api/qcFlags.test.js [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+        it('should successfuly filter on createdBy', async () => {
+            const response = await request(server).get('/api/qcFlags?filter[createdBy][]=John%20Doe');
+            expect(response.status).to.be.equal(200);
+            const { data } = response.body;
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(3);
+            expect(data.map(({ id }) => id)).to.have.all.members([1, 2, 3]);
+        });
+        it('should support sorting by id', async () => {
+            const response = await request(server).get('/api/qcFlags?sort[id]=ASC');
```
Comment (martinboulais) [inline test/api/qcFlags.test.js:127]: If you test sorting, you should test both ways. 
Or one way here but both ways in the service?

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: test/api/qcFlags.test.js (hunk 1 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js]: removed

File: test/api/qcFlags.test.js (hunk 5 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:71]: applied

File: test/api/qcFlags.test.js (hunk 4 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:57]: done

File: test/api/qcFlags.test.js (hunk 2 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:29]: done

File: test/api/qcFlags.test.js (hunk 3 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:32]: done

File: test/api/qcFlags.test.js (hunk 8 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:127]: tests added

File: test/api/qcFlags.test.js (hunk 6 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:64]: applied, checking one full object left, there is not other check for whether the structure is correct

File: test/api/qcFlags.test.js (hunk 7 above)
Comment (xsalonx) [reply test/api/qcFlags.test.js:113]: renamed

**Pull request merged**